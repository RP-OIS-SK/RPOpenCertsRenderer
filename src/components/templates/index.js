// This file is the template registry serves as the template registry.
// When templates are loaded, their name will be compared to the keys of the
// exported object and that template will be used. If a template cannot be
// found, the default template will be used instead.

// TODO: Use dynamic loading to prevent all templates to be loaded at once.
import DefaultTemplate from "./default";
import CustomTemplate from "./customTemplate";
import GovTechDemoCert from "./govtechDemoCert";
import RP_2020_P_MAIN from "./RPTemplate/2020-P-MAIN";
import RP_2020_P_CCA from "./RPTemplate/2020-P-CCA";

export default {
  default: DefaultTemplate,
  CUSTOM_TEMPLATE: CustomTemplate,
  GOVTECH_DEMO: GovTechDemoCert,
  RP_2020_P_MAIN: RP_2020_P_MAIN,
  RP_2020_P_CCA: RP_2020_P_CCA,
  NULL: []
};
