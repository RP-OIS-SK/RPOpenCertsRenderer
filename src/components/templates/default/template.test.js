import React from "react";
import renderer from "react-test-renderer";
import templates from "./index";
import Template from "./template";

describe("index", () => {
  it("exports a templates array", () => {
    expect(Array.isArray(templates)).toBe(true);
    templates.forEach(template => {
      expect(template.id).toBeTruthy();
      expect(template.label).toBeTruthy();
      expect(typeof template.template).toBe("function");
    });
  });
});

describe("default template", () => {
  it("matches snapshot", () => {
    const document = { foo: "bar" };
    const component = renderer.create(<Template document={document} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
