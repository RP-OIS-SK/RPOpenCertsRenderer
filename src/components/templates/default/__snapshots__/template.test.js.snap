// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default template matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="borders"
  >
    <div
      className="borders"
    >
      <div
        className="p-4 background"
        data-component-id="toggle-display"
        onClick={[Function]}
      >
        foo
      </div>
      <div
        className="p-4 text-truncate "
        data-component-id="displayed-value"
      >
        bar
      </div>
    </div>
  </div>
</div>
`;
